warp10/src/main/java/io/warp10/WarpDist.java

```diff
@@ -345,4 +307,19 @@ public static boolean isEgress() {
   public static void setEgress(boolean egress) {
     hasEgress = egress;
   }
+
+  public static void extractKeys(KeyStore keystore, Properties properties) {
+    KeyStore.checkAndSetKey(keystore, KeyStore.SIPHASH_CLASS, properties, Configuration.WARP_HASH_CLASS, 128);
+    KeyStore.checkAndSetKey(keystore, KeyStore.SIPHASH_LABELS, properties, Configuration.WARP_HASH_LABELS, 128);
+    KeyStore.checkAndSetKey(keystore, KeyStore.SIPHASH_TOKEN, properties, Configuration.WARP_HASH_TOKEN, 128);
+    KeyStore.checkAndSetKey(keystore, KeyStore.SIPHASH_APPID, properties, Configuration.WARP_HASH_APP, 128);
+    KeyStore.checkAndSetKey(keystore, KeyStore.AES_TOKEN, properties, Configuration.WARP_AES_TOKEN, 128, 192, 256);
+    KeyStore.checkAndSetKey(keystore, KeyStore.AES_SECURESCRIPTS, properties, Configuration.WARP_AES_SCRIPTS, 128, 192, 256);
+    KeyStore.checkAndSetKey(keystore, KeyStore.AES_METASETS, properties, Configuration.WARP_AES_METASETS, 128, 192, 256);
+    KeyStore.checkAndSetKey(keystore, KeyStore.SIPHASH_CLASS, properties, Configuration.WARP_HASH_CLASS, Configuration.WARP_DEFAULT_AES_LOGGING, 128, 192, 256);
```

Comment:
Incorrect parameters.

Reply:
Nice catch. I went again through the changes and it's now highly unlikely that there is another error of this kind.